frontend/src/components/input-output/configure-ds/ConfigureDs.jsx

```diff
@@ -50,9 +50,12 @@ function ConfigureDs({
   }, [formData]);
 
   useEffect(() => {
-    if (connDetails && connDetails.connector_id !== selectedSourceId) {
+    if (
+      connDetails?.connector_id &&
+      connDetails?.connector_id !== selectedSourceId
+    ) {
       setFormData({});
-    } else {
+    } else if (metadata) {
       setFormData(metadata);
     }
   }, [selectedSourceId]);
```

Comment:
@kirtimanmishrazipstack This useEffect hooks seems to be a bit confusing. I have suggestion for optimizing the code and making it more readable. Can you please refer the below and see if it works?

cc: @vishnuszipstack 

```
useEffect(() => {
    const { connector_id: connectorId } = connDetails || {};

    // Check if connectorId matches selectedSourceId and metadata is available
    const shouldSetMetadata = connectorId === selectedSourceId && metadata;

    // Set formData based on the conditions
    setFormData(shouldSetMetadata ? metadata : {});

  }, [selectedSourceId]);
```